### PR TITLE
[CPU Plugin 2.0] replace ie::precision with ov::element

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - remove IE::Precision in CPU plugin
 - replace with ov::element

### Tickets:
 - 124100

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: src/plugins/intel_cpu/src/memory_desc/cpu_memory_desc_utils.cpp [hunk 1]
```diff
@@ -77,8 +83,13 @@ DnnlBlockedMemoryDesc MemoryDescUtils::convertToDnnlBlockedMemoryDesc(const Infe
         std::fill(strides.begin(), strides.end(), 0);
     }
 
-    return DnnlBlockedMemoryDesc(desc.getPrecision(), Shape(desc.getDims()), blkDesc.getBlockDims(), blkDesc.getOrder(), blkDesc.getOffsetPadding(),
-                                 blkDesc.getOffsetPaddingToData(), strides);
+    return DnnlBlockedMemoryDesc(InferenceEngine::details::convertPrecision(desc.getPrecision()),
```
Comment (vurusovs) [inline src/plugins/intel_cpu/src/memory_desc/cpu_memory_desc_utils.cpp:86]: I see use of `DnnlBlockedMemoryDesc` with mixed IE precision and ov element types in different places. With current implementation function operates with ov element types. Please, recheck

File: src/tests/functional/plugin/shared/src/execution_graph_tests/runtime_precision.cpp [hunk 2]
```diff
@@ -123,7 +123,23 @@ TEST_P(ExecGraphRuntimePrecision, CheckRuntimePrecision) {
         if (rtIter == rtInfo.end())
             FAIL() << "Runtime precision is not found for node: " << opIter->get()->get_friendly_name();
 
-        if (expectedPrc.second.name() != rtIter->second.as<std::string>())
+        bool success_status = false;
+        success_status = std::regex_match(expectedPrc.second.name(),
```
Comment (vurusovs) [inline src/tests/functional/plugin/shared/src/execution_graph_tests/runtime_precision.cpp:127]: Solution looks overcomplicated. Can it be simplified by using some converters like `convertPrecision` etc.?

**Review by vurusovs — CHANGES_REQUESTED** [review]

File: src/plugins/intel_cpu/src/nodes/transpose.cpp [hunk 3]
```diff
@@ -111,7 +111,7 @@ void Transpose::initSupportedPrimitiveDescriptors() {
             supportedPrimitiveDescriptorsBuilder(config, transposeParams);
         }
 #endif // OPENVINO_ARCH_X86_64
-        if (prec == ov::element::f32 || prec == ov::element::f16 || prec == ov::element::i8 || prec == ov::element::u8) {
+        if (prec == ov::element::f32 || prec == ov::element::f16 || prec == ov::element::i8 || prec == ov::element::u8 || prec == ov::element::bf16) {
```
Comment (riverlijunjie) [inline src/plugins/intel_cpu/src/nodes/transpose.cpp:114]: Why add `bf16` condition?

File: src/plugins/intel_cpu/src/memory_desc/cpu_memory_desc_utils.cpp (hunk 1 above)
Comment (riverlijunjie) [reply src/plugins/intel_cpu/src/memory_desc/cpu_memory_desc_utils.cpp:86]: I think we need decouple precision and MemoryDesc, we have another PR to solve TensorDesc issue(https://github.com/openvinotoolkit/openvino/pull/21002), the second merged PR will solve this mixed issue.

File: src/plugins/intel_cpu/src/nodes/transpose.cpp (hunk 3 above)
Comment (zhaixuejun1993) [reply src/plugins/intel_cpu/src/nodes/transpose.cpp:114]: The master had this "Precision::BF16"
https://github.com/openvinotoolkit/openvino/blob/369831caef55efe1f2d87889d3997f93c53ab6b8/src/plugins/intel_cpu/src/nodes/transpose.cpp#L114

File: src/plugins/intel_cpu/src/memory_desc/cpu_memory_desc_utils.cpp (hunk 1 above)
Comment (zhaixuejun1993) [reply src/plugins/intel_cpu/src/memory_desc/cpu_memory_desc_utils.cpp:86]: Agree with @riverlijunjie, will clear this after PR (https://github.com/openvinotoolkit/openvino/pull/21002)

**Review by riverlijunjie — APPROVED** [review]

**Review by vurusovs — APPROVED** [review]

**Pull request merged**